src/os/Transaction.cc

```diff
@@ -232,6 +232,25 @@ void ObjectStore::Transaction::_build_actions_from_tbl()
       }
       break;
 
+    case Transaction::OP_MERGE_DELETE:
+      {
+	coll_t temp_cid;
```

Comment:
Actually, this shouldn't be encoding the source_cid separately at all.  The source object must be in the same collection (ObjectStore level).  The *filestore* might use the temp version of the collection instead, but that depends on the ghobject_t passed, not on the collection.